### ISSUE TITLE
Best settings for serving others

### ISSUE THREAD

**Issue by andrasfuchs** [BiglySoftware/BiglyBT#2902]
First of all, @parg, I would like to thank you, your work means a lot to a lot of people. Thank you for maintaining and continuously improving BiglyBT!

This might be a naïve question, but are there any settings that would enable me/my computer to serve others better? It would be OK with me if BiglyBT used 90% of my disk space and upload bandwidth all the time, or if I had a bigger distributed DB, etc., if I knew that this helps others to get their torrents quicker or if I could keep needed torrents alive (even if I was not interested in those particular torrents).

What would be the best strategy to achieve this goal and how could BiglyBT help this goal? Is there an automated way to find, download and seed torrents that are "underserved"?

I'm also curious if [IPFS](https://ipfs.tech/) fits into this picture somehow? Is there any cross section between IPFS and torrents (v2)? Can I serve my files on them both?

**Comment by parg** [comment]
Thanks for the kind words!

I don't have any magic settings to tell you to manage BiglyBT better - it depends what you are trying to achieve. Some users maintain extensive libraries of rare content so they can serve the occasional peer that comes along. So they might not in general have high upload rates but keep stuff alive. Others on private trackers do everything they can to maintain their share ratio, downloading stuff they don't even want in an attempt to upload more than they download.

If you can live with high CPU usage by BiglyBT then set up the queue rules to allow a large number of downloads to be actively running. While BiglyBT will cycle through queued downloads over time you're more likely to find peers to upload to with more active. Obviously there's a limit there that depends on your computing power, 100,000 active downloads will probably end up with BiglyBT spending all its time thrashing around processing them rather than delivering to the network.

Upload slots limit the number of peers that BiglyBT will upload to at any time so, for downloads with lots of leechers, increasing them can help. 

Overall though, older content is hard to upload as there are few leechers and many seeds, so  you have to live with that.

There has been some interest in integrating IPFS with BiglyBT but at the moment I'm not convinced that closer ties between the two would be worth the effort. I would assume that if you have completed files in BiglyBT then you can also share them over IPFS, you'd just need to manually manage the process. Doing the opposite - taking content from IPFS and somehow constructing torrents out of them wouldn't be worth doing as there would be nobody out there with your nice new torrent files. Searching IPFS for content in a torrent file you were downloading and then using that to fill in your BiglyBT download is possible but what are the chances that the content is there?

**Comment by andrasfuchs** [comment]
Thank you for the detailed reply, I appreciate it!

I started to play around with the settings, and I ended up setting up an RSS feed from Jackett that has the potential torrents that I would like to seed. It looks really good and I love the fact the it gets refreshed automatically and I can set them to be auto-downloaded!

![image](https://user-images.githubusercontent.com/910321/223719844-d2486fb2-bfbe-4b74-b628-0d5bae065cc4.png)

The only issue I have with this method, that I can't filter the list based on the max seeds and min peers. Ideally I could set it up in a way that only torrent with seeders between 3 and 20, and with peers of 1 or higher gets downloaded in the end.

As a feature request, would you consider adding the "Max Seeds" and "Min Peers" filter to the Subscription dialog?

**Comment by parg** [comment]
Sure

**Comment by parg** [comment]
Check out 3301 beta B15

**Comment by andrasfuchs** [comment]
Thank you, I have been using these settings in the last 10 days. They work great, my bandwidth was used constantly, I could seed many torrents that were in need. 

![image](https://user-images.githubusercontent.com/910321/226143314-eb2344be-1edf-42d9-adae-7c8f66bfe9c3.png)

I used all the subscription settings available (min/max size, min/max seeds, min peers, max age) and I set the `Auto-download new results` and `auto-tagging` function as well.

I noticed one thing: these torrents start to accumulate a little too much over time. Would it be technically possible to not just only auto-download new results, but also delete those that are no longer satisfy the filters in the subscription that they were downloaded by? 

For example, if a subscription has min seeds=3 and max seeds=5, and one of its torrents gets auto-downloaded with 3 seeds initially, but over time it gets to the point where it has 6 seeds, then it would be deleted automatically.

This new feature would make the "donation scenario" that I have fully automated.

Thank you for considering!
